controllers/managedocs_controller.go

```diff
@@ -210,21 +217,29 @@ func (r *ManagedOCSReconciler) setStorageClusterParamsFromSecret(sc *ocsv1.Stora
 	if err != nil {
 		return fmt.Errorf("Invalid storage cluster size value: %v", sizeAsString)
 	}
-
 	r.Log.Info("Storage cluster parameters", "count", sdsCount)
 
 	var ds *ocsv1.StorageDeviceSet = nil
-	for _, item := range sc.Spec.StorageDeviceSets {
+	for index, item := range sc.Spec.StorageDeviceSets {
 		if item.Name == deviceSetName {
-			ds = &item
+			ds = &sc.Spec.StorageDeviceSets[index]
 			break
 		}
 	}
 	if ds == nil {
 		return fmt.Errorf("cloud not find default device set on stroage cluster")
 	}
 
-	sc.Spec.StorageDeviceSets[0].Count = sdsCount
+	// Prevent downscaling by comparing count from secret and count from storage cluster
+	if currSdsCount <= sdsCount {
+		ds.Count = sdsCount
+		r.Log.Info("Actual storage cluster parameter", "actualCount", ds.Count)
+	} else {
+		ds.Count = currSdsCount
+		err := fmt.Errorf("Cannot set storage device set count: actualCount=%v", ds.Count)
+		r.Log.Error(err, "Reducing storage device set count is not allowed")
```

Comment:
This condition should not result in aborting the reconcile loop, as such we should log it as a warning and not an error. In this case, there isn't any real reason to create the `error` object at all.